iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/SectionModification.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.modification;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.ShuntCompensator;
+
+import java.util.Objects;
+
+/**
+ * Simple {@link NetworkModification} to change the section of a shunt compensator.
+ *
+ * @author Nicolas PIERRE <nicolas.pierre at artelys.com>
+ */
+public class SectionModification extends AbstractNetworkModification {
+
+    private final String shuntCompensatorId;
+    private final int sectionCount;
+
+    public SectionModification(String shuntCompensatorId, int sectionCount) {
+        this.shuntCompensatorId = Objects.requireNonNull(shuntCompensatorId);
+        this.sectionCount = sectionCount;
+        if (sectionCount < 0) {
+            throw new PowsyblException("Section count of a Shunt can't be negative");
+        }
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager,
+                      Reporter reporter) {
+        ShuntCompensator shuntCompensator = network.getShuntCompensator(shuntCompensatorId);
+        if (shuntCompensator == null) {
+            logOrThrow(throwException, "Shunt Compensator '" + shuntCompensatorId + "' not found");
+            return;
+        } else if (shuntCompensator.getMaximumSectionCount() < sectionCount) {
```

Comment:
Same comment, this check already exists. I am not in favor of code duplication. @olperr1 what should we do?

Reply:
This check exists in `setSectionCount` (same place as the check ` > 0`). It will be cleaner if it is removed in this `apply` method.